src/enums.rs

```diff
@@ -100,10 +100,22 @@ pub enum ReedlineEvent {
     /// Trigger Tab
     HandleTab,
 
-    /// Don't know a better name for this
+    /// Handle EndOfLine event
+    ///
+    /// Expected Behavior:
+    ///
+    /// - On empty line breaks execution to exit with [`Signal::CtrlD`]
+    /// - Secondary behavior [`EditCommand::Delete`]
     CtrlD,
```

Comment:
The ASCII control sequence names would be (linked to the ctrl+letter combination in encoding) End of text or ETX for `Ctrl+C` and End of transmission EOT for `Ctrl+D`. But the signals would typically be interpreted as semantically something slightly different (e.g. SIGTERM, EOF)

Reply:
What's your take on the names that we should go with then?
- `CtrlC` and `CtrlD`
- `EndOfText` and `EndOfTransmission`
- `Sigterm` and `Eof`

The last one is how we are supposed to interpret so that makes the most sense. Thoughts?

Reply:
IIRC, I think `CtrlC` sends a `SIGINT`. However, I'm not a big fan of these sig names because it presumes the reader of the source code fully understands them. To complicate it further, Windows does not handle signal processing the way Unix does. Not that we're exactly talking about software interrupts and signal processing, but you get my meaning, I think. My preference, FWIW, is that we should strive for clarity for the novice because 'naming is hard', as jt likes to say, and we want to make our intentions clear in our code.

Reply:
Just to throw a couple more in: `EndOfInput` and `ExitShell`